src/java/org/apache/cassandra/metrics/HintsServiceMetrics.java

```diff
@@ -51,6 +62,12 @@ public final class HintsServiceMetrics
                                                                                                .executor(ImmediateExecutor.INSTANCE)
                                                                                                .build(address -> Metrics.histogram(factory.createMetricName("Hint_delays-"+address.toString().replace(':', '.')), false));
 
+    public HintsServiceMetrics(HintsService hintsService)
+    {
+        hintsFileSize = Metrics.register(factory.createMetricName("HintsFileSize"),
```

Comment:
It seems like the only purpose of putting `hintsService` into the constructor is to get access to `getTotalHintsSizeOfNode` on it. We might go with parameter-less constructor instead and do `HintsService.instance::getTotalHintsSizeOfNode()`